### PR TITLE
fix: Add envelope to `CuboidVolumeBuilder`

### PR DESCRIPTION
**Pull request by andiwand**
I noticed that the `CuboidVolumeBuilder` is not generating any envelope around the generated layer which results in overlapping surfaces which does not play nicely with navigation.

discovered in https://github.com/acts-project/acts/pull/2532

### CODE REVIEW

File: Core/src/Geometry/CuboidVolumeBuilder.cpp [hunk 1]
```diff
@@ -169,6 +169,9 @@ std::shared_ptr<Acts::TrackingVolume> Acts::CuboidVolumeBuilder::buildVolume(
 
     LayerConfig lCfg;
     lCfg.surfaceCfg = {sCfg};
+    lCfg.envelopeX = {0.1 * UnitConstants::mm, 0.1 * UnitConstants::mm};
+    lCfg.envelopeY = {0.1 * UnitConstants::mm, 0.1 * UnitConstants::mm};
+    lCfg.envelopeZ = {0.1 * UnitConstants::mm, 0.1 * UnitConstants::mm};
```
Comment (felix-russo) [inline Core/src/Geometry/CuboidVolumeBuilder.cpp:174]: Why were these values chosen?

**Review by felix-russo — COMMENTED** [review]
Looks good, just one question for my understanding

**Review by paulgessinger — APPROVED** [review]

File: Core/src/Geometry/CuboidVolumeBuilder.cpp (hunk 1 above)
Comment (andiwand) [reply Core/src/Geometry/CuboidVolumeBuilder.cpp:174]: mostly random. they need to be big enough to not collide with the surface tolerance but small enough to not collide with the geometry

**Comment by paulgessinger** [comment]
Actually, I just now remembered https://github.com/acts-project/acts/pull/1751/. So shouldn't this in principle already be configurable (but not correctly configured)?

**Comment by andiwand** [comment]
> Actually, I just now remembered #1751. So shouldn't this in principle already be configurable (but not correctly configured)?

right it is configurable but then we default it in the volume builder which results in 0 envelope again

**Comment by paulgessinger** [comment]
@andiwand Ah, and this is only if the cuboid builder auto-builds a single layer, without an explicit layer configuration.

**Review by paulgessinger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33147089) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33147088) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33147087) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33147086) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33147085) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33147084) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33147083) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33147082) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33147081) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33147080) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33147079) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33147078) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33147077) |
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33147076) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33147075) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33147074) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33147073) |

**Pull request merged**